Add new icons, adopta logo and update styles to match PUC style guide

jasonlally opened the pull request:
This PR implements the icons and adopt a logo developed by PUC. It also has some minor structural changes to enable differentiating amongst drains adopted by the logged in user.

I've also updated button colors and other elements to use colors from the PUC style guide.
[STYLE GUIDE REVISED 7.13.pdf](https://github.com/sfbrigade/adopt-a-drain/files/60873/STYLE.GUIDE.REVISED.7.13.pdf)

@jszwedko - I had done a little to start cleaning up form error messages when I realized you already had a PR on that #72  I went ahead and merged them into this one.

jszwedko commented on app/assets/javascripts/main.js.erb:
```diff
@@ -706,10 +764,11 @@ $(function() {
           data: {
-            'thing_id': activeThingId,
+            'thing_id': activeThingId
+            /*
```
Can you throw some `*`s on the commented lines just to make it stand out more that those lines are commented?

jszwedko commented on app/assets/javascripts/main.js.erb.orig:
```diff
@@ -0,0 +1,881 @@
+    if(!$(this).data('state') || $(this).data('state') === 'user_sign_up') {
+      if($('#user_name').val() === '') {
+        errors.push($('#user_name'));
+<<<<<<< HEAD
```
Looks like you have some unresolved merge conflicts.

jszwedko commented on app/assets/javascripts/main.js.erb:
```diff
@@ -59,16 +71,48 @@ $(function() {
+      name: 'Adopted',
+      icon: iconBase + 'adopted.png'
+    },
+    adopted_by_you: {
```
:+1: for differentiating between adopted drains and _your_ adopted drains.

jszwedko commented on app/assets/javascripts/main.js.erb:
```diff
@@ -59,16 +71,48 @@ $(function() {
+
+  var legend = document.getElementById('legend');
+  var iconBase = '/assets/markers/';
+  console.log(iconBase);
```
Probably want to drop this before mergingi.

jszwedko commented on app/assets/javascripts/main.js.erb:
```diff
@@ -468,16 +525,17 @@ $(function() {
-              'thing_id': activeThingId,
+              'thing_id': activeThingId
+              /*
               'flash': {
```
Same as below, could you add `*`s to the commented lines to visually make them stand out?

jszwedko commented on app/assets/javascripts/main.js.erb.orig:
```diff
@@ -0,0 +1,881 @@
+$(function() {
```
Just realized this whole file is the result of merge conflicts, it may be sufficient to just drop this one.

jszwedko commented on app/assets/stylesheets/application.css:
```diff
@@ -15,22 +15,32 @@
+ }
 
-img[title="Code for San Francisco"]{
+ img[title="Code for San Francisco"]{
```
Is this leading whitespace (and the leading whitespace above) needed?

jasonlally replied:
hmmmm...weird, I don't know why my editor indented those, fixing this

jszwedko commented on app/controllers/things_controller.rb:
```diff
@@ -3,10 +3,21 @@ class ThingsController < ApplicationController
+    end
+  end
+
+  def check_owner(thing)
```
I think this method better belongs on the model, something like:

```
def is_owned_by?(owner)
  return false unless owner

  return self.user.id == owner.id
end
```

Just to adhere to the "fat model, skinny controller" pattern.

jszwedko replied:
It occurred to me that this logic could also be moved to the Javascript since the result set will include the user id and you should have access to that there (or we can inject it so you do).

jasonlally replied:
I think you're right, I'm investigating this option now.

jszwedko commented on app/models/thing.rb:
```diff
@@ -11,6 +11,7 @@ class Thing < ActiveRecord::Base
   validates :city_id, uniqueness: true, allow_nil: true
   validates :lat, presence: true
   validates :lng, presence: true
+  attr_accessor :owned_by_you
```
It feels a little weird to have this attribute exist here, but only be populated by the controller. This'll probably be mitigated by moving the method for checking the owner into here.

jszwedko commented on app/views/sidebar/_combo_form.html.haml.orig:
```diff
@@ -0,0 +1,67 @@
+= form_for :user, :html => {:id => "combo-form", :class => "form-vertical"} do |f|
```
Looks like another leftover from a merge conflict.

jasonlally replied:
Ah! Sorry!

jszwedko commented on app/views/sidebar/_search.html.haml.orig:
```diff
@@ -0,0 +1,16 @@
+= form_tag "/address", :method => "get", :id => "address_form", :class => "search-form" do
```
More merge conflict vestiges :smile:

jszwedko commented:
Overall :+1: :+1:, thanks for obtaining and implementing the new assets!

I'm going to merge #72 just so this diff is a little more manageable to review.

jszwedko commented:
Also :clap: for style guide, I put a link to it on the wiki.

jasonlally commented:
@jszwedko I returned the Things model and controller to its prior state and implemented the adopted by you on the client side.

I don't think I've added any new lines (after returning to the prior state), but I'm getting a decrease in coverage. It looks like the error states aren't being tested.

Can you take another look? I didn't perform a manual merge, but I pushed my changes here. I think all of the conflicts are resolved. Sorry about this messy branch.

jszwedko commented:
No worries! I rebased this branch against production and opened #76 to get this mergeable. Want to give that one a once over? I'll check the coverage of it.

jszwedko commented:
Also :clap: for doing the icon piece on the client side -- I think that is much cleaner.

jszwedko commented:
Closed by #76
